### PR TITLE
ep_webrtc.enabled -> ep_webrtc.disabled

### PR DESCRIPTION
**Pull request by orblivion**
Previously I created the audio and video on/off switch per user. It had these notable attributes:
* Using `settingToCheckbox`
* Using `disabled` rather than `enabled` as the `settings.json` field name
* Having a little more clear URL var option than `av=YES`

That has left the existing `ep_webrtc.enabled` variable without the above attributes, so I'm applying them here as `ep_webrtc.disabled`.
* I'm leaving `ep_webrtc.enabled` usable with a deprecation warning. If you use both you get a config error.
* `av=YES` no longer works. So far this doesn't have smooth deprecation (@JohnMcLear lmk if I should smoothly deprecate it).
* Existing user-based setting should hold during the transition. I tested this pretty carefully.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -157,22 +152,17 @@ exports.setSocketIO = function (hook, context, callback)
 
 exports.eejsBlock_mySettings = function (hook, context, callback)
 {
-    var enabled = (settings.ep_webrtc && settings.ep_webrtc.enabled === false)
-      ? 'unchecked'
-      : 'checked';
-
```
Comment (orblivion) [inline index.js:163]: Not setting the initial value here since `settingToCheckbox` takes care of it.

File: index.js [hunk 2]
```diff
@@ -190,26 +180,56 @@ exports.eejsBlock_styles = function (hook_name, args, cb) {
 };
 
 function validateSettings() {
```
Comment (orblivion) [inline index.js:182]: Note that returning `false` with this function gives a gritter error telling users to alert the admin that they should see an error in their logs. It doesn't tell the users what the error is.

File: static/tests/frontend/specs/interface_buttons.js [hunk 3]
```diff
@@ -63,21 +63,28 @@ describe('Test the behavior of the interface buttons: Mute, Video Disable, Enlar
 
       this.timeout(60000);
 
-      expect(chrome$("video").css("width")).to.be("160px");
```
Comment (orblivion) [inline static/tests/frontend/specs/interface_buttons.js:66]: This is an unrelated test failure that appeared randomly. It was getting back a value of `160.25px` for whatever reason. I'm just gonna round to the nearest integer and hope for the best going forward.

File: templates/settings.ejs [hunk 4]
```diff
@@ -1,5 +1,5 @@
 <p>
-  <input type="checkbox" id="options-enablertc" <%= enabled %>></input>
```
Comment (orblivion) [inline templates/settings.ejs:2]: https://github.com/ether/ep_webrtc/pull/65/files#r463879287

**Comment by JohnMcLear** [comment]
Looks good but conflicts galore, can you resolve the conflicts please so I can rebase?

Noted RE av=Yes, I think I can move my client over to the new URL param, will chase on that

**Comment by packardone** [comment]
> Looks good but conflicts galore, can you resolve the conflicts please so I can rebase?
> 
> Noted RE av=Yes, I think I can move my client over to the new URL param, will chase on that

av=YES actually does something different, and that functionality shouldn't be lost.

As I understand it, if webrtcaudioenabled=false is in the URL, webrtc starts, but audio is muted on load.  If webrtcvideoenabled=false is in the URL, webrtc starts but video is muted on load.  If both are set to false, then video and audio are both muted.  However, a user can still un/mute either using the options in the webrtc display.  This is certainly desired behavior.

However, av=YES was more of a master switch.  It turned on/off the entire webrtc plugin for a given pad.  In other words, if av=NO was set, individuals wouldn't see the webrtc column at all or have access to any of the webrtc options.  If av=YES was set, the webrtc column/functions would appear.  We would need to maintain this functionality.

I hope that this helps!

**Comment by rhansen** [comment]
Superseded by commit f8db3b979c4786b6045b15cc5c7a8bf2d455daac.

**Pull request closed without merging**